executor/explain.go

```diff
@@ -70,23 +71,40 @@ func (e *ExplainExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	return nil
 }
 
-func (e *ExplainExec) generateExplainInfo(ctx context.Context) ([][]string, error) {
+func (e *ExplainExec) generateExplainInfo(ctx context.Context) (rows [][]string, err error) {
+	closed := false
+	defer func() {
+		// handle panic
+		if panicErr := recover(); panicErr != nil && !closed {
+			err = e.analyzeExec.Close()
+			closed = true
+		}
```

Comment:
If you call `recover` here, the panic will be recovered here, which is not we want.
Suggest just check `closed`.

```go
defer func() {
    if !closed { e.analyzeExec.Close() }
}()

Reply:
if without recover, the goroutine will panic, and test will not work.